src/lib/containers/QuerySortSelector.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from 'react'
+import { SortConfiguration } from './CardContainerLogic'
+import { useInfiniteQueryContext } from './QueryContext'
+import { SortDirection, SortItem } from '../utils/synapseTypes'
+import { unCamelCase } from '../utils/functions/unCamelCase'
+import Typography from '../utils/typography/Typography'
+import Creatable from 'react-select/creatable'
+import {
+  EnumOption,
+  findValueOption,
+  Control,
+} from './entity/annotations/CustomSelectWidget'
+
+export type QuerySortSelectorProps = {
+  sortConfig: SortConfiguration
+  facetAliases?: Record<string, string>
+}
+
+const QuerySortSelector: React.FunctionComponent<QuerySortSelectorProps> = ({
+  sortConfig,
+  facetAliases,
+}) => {
+  const { defaultColumn, defaultDirection, sortableColumns } = sortConfig
+  const infiniteQueryContext = useInfiniteQueryContext()
+  const { getLastQueryRequest, executeQueryRequest } = infiniteQueryContext
+  const [sortColumn, setSortColumn] = useState<string | undefined>(
+    defaultColumn,
+  )
+  const [sortDirection, setSortDirection] = useState(defaultDirection)
+
+  const enumOptions: EnumOption[] = sortableColumns.map(sortableColumn => {
+    return {
+      value: sortableColumn,
+      label: unCamelCase(sortableColumn, facetAliases)!,
+    }
+  })
+
+  const onChange = (value?: string) => {
+    const lastQueryRequestDeepClone = getLastQueryRequest()
+    let newSortDirection: SortDirection = 'ASC'
+    if (value === sortColumn) {
+      // flip sort direction
+      newSortDirection = sortDirection === 'ASC' ? 'DESC' : 'ASC'
+    }
+
+    const newSortItems: SortItem[] | undefined = value
+      ? [
+          {
+            column: value,
+            direction: newSortDirection,
+          },
+        ]
+      : undefined
+    lastQueryRequestDeepClone.query.sort = newSortItems
+    executeQueryRequest(lastQueryRequestDeepClone)
+    setSortColumn(value)
+    setSortDirection(newSortDirection)
+  }
+
+  return (
+    <div className="QuerySortSelector bootstrap-4-backport">
+      <Typography variant="label" className="sort-by-label SRC-inlineBlock">
+        Sort by
+      </Typography>
+      <Creatable
```

Comment:
Don't use `Creatable`; we use it in the annotations editor because in that scenario the user can set a value outside of the enum. Not true here (right?), so instead use `Select`

Reply:
You're right.  I started with Select, ran into all kinds of trouble with the types, so I changed it to utilize the same solutions from the annotations editor.  But I forgot to change it back to a Select.

Reply:
Changed to be a `Select`